### PR TITLE
(DOCSP-13361): Add third-party license page

### PR DESCRIPTION
**Pull request by cbush**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCS-13361

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/third-party-license/reference/third-party-licenses.html

### CODE REVIEW

File: source/reference/third-party-licenses.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+{+service+} uses third-party libraries or other resources that may be
+distributed under licenses different than the MongoDB software.
+{+service-short+} depends upon the following third-party packages. These
+packages are licensed as shown in the following list.
```
Comment (mongomoe) [inline source/reference/third-party-licenses.txt:21]:
```suggestion
packages are licensed, as shown in the following list.
```

File: source/reference/third-party-licenses.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+{+service+} uses third-party libraries or other resources that may be
```
Comment (mongomoe) [inline source/reference/third-party-licenses.txt:18]: 'may be distributed' is passive voice bc the sentence doesn't define who is doing the distribution.

File: source/reference/third-party-licenses.txt (hunk 1 above)
Comment (cbush) [reply source/reference/third-party-licenses.txt:21]: That would change the meaning of the sentence from what was intended.

File: source/reference/third-party-licenses.txt (hunk 2 above)
Comment (cbush) [reply source/reference/third-party-licenses.txt:18]: True but in a dependent clause that's not so bad. (Plus inherited this text from CET)

**Pull request merged**